rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -514,5 +550,30 @@ object Reduce {
       val thread = Thread.currentThread.getName
       println(s"$now [$thread]" + "\n" + msg)
     }
+
+    val methodTable: Map[String, ((Par, Seq[Par]) => (Env[Par] => Task[Par]))] =
+      Map("nth" -> {
+        def localNth(ps: Seq[Par], nth: Int): Task[Par] =
+          if (ps.isDefinedAt(nth))
+            Task.pure(ps(nth))
+          else
+            Task raiseError new Error("Error: index out of bound: " + nth)
+        (p: Par, args: Seq[Par]) => (env: Env[Par]) =>
+          {
+            if (args.length != 1)
+              Task raiseError new Error("Error: nth expects 1 argument")
+            for {
+              nth <- evalToInt(args(0))(env)
+              v   <- evalSingleExpr(p)(env)
+              result <- v.exprInstance match {
+                         case EListBody(EList(ps, _, _, _))   => localNth(ps, nth)
+                         case ETupleBody(ETuple(ps, _, _, _)) => localNth(ps, nth)
+                         case _ =>
+                           Task raiseError new Error(
+                             "Error: nth applied to something that wasn't a list or tuple.")
```

Comment:
Are there plans to take more than either a list or tuple? It seems like just taking either would suffice.

Reply:
I did not have plans to make the nth method specifically handle anything but tuples and lists.
Other methods may support additional kinds of data.
For instance, Mike has plans for us to implement confinement as a method on arbitrary processes.

Reply:
Got it